packages/grafana-data/src/types/ScopedVars.ts

```diff
@@ -1,8 +1,24 @@
+import { DataFrame, Field } from './dataFrame';
+import { DisplayValue } from './displayValue';
+
 export interface ScopedVar<T = any> {
   text?: any;
   value: T;
-  skipUrlSync?: boolean;
-  skipFormat?: boolean;
 }
 
-export interface ScopedVars extends Record<string, ScopedVar> {}
+export interface ScopedVars {
+  __dataContext?: DataContextScopedVar;
+  [key: string]: ScopedVar | undefined;
+}
+
+/**
+ * Used by data link macros
+ */
+export interface DataContextScopedVar {
+  value: {
+    frame: DataFrame;
+    field: Field;
+    valueIndex?: number;
+    calculatedValue?: DisplayValue;
+  };
+}
```

Comment:
I wish this was just fieldIndex instead of the field, but signature for getLinksSupplier does not include fieldIndex just field,  and getLinksSupplier is exported and use in a few places. Could calculate/try to find the fieldIndex